### PR TITLE
Xplat script improvements

### LINKED ISSUE #628: cibuild.sh should fail if init-tools.sh fails
I was having some network issues on my Mac and got this:

```
Installing dotnet cli...
curl: (56) SSLRead() return error -9806
tar: Error opening archive: Failed to open '/Users/raines/src/msbuild/Tools/dotnetcli/dotnet.tar'
Restoring BuildTools version 1.0.25-prerelease-00231-07...
./init-tools.sh: line 78: /Users/raines/src/msbuild/Tools/dotnetcli/dotnet: No such file or directory
ERROR: Could not restore build tools correctly. See '/Users/raines/src/msbuild/init-tools.log' for more details.
Initializing BuildTools...
./init-tools.sh: line 84: /Users/raines/src/msbuild/packages/Microsoft.DotNet.BuildTools/1.0.25-prerelease-00231-07/lib/init-tools.sh: No such file or directory
Done initializing tools.

** Rebuilding MSBuild with binaries from BuildTools

** using MSBuild in path: ./Tools/MSBuild.exe
** using runtime host in path: ./Tools/corerun
** ./Tools/corerun  ./Tools/MSBuild.exe "./build.proj" /t:Build /p:OS=OSX /p:Configuration=Debug-NetCore /verbosity:minimal  /m /fl "/flp:v=diag;logfile=./msbuild_bootstrap_build.log"
./cibuild.sh: line 57: ./Tools/corerun: No such file or directory
```

The script should have early-aborted as soon as curl failed.

### PR DESCRIPTION
**Pull request by rainersigwald**
A few script fixes:

* Use the just-restored `corerun` for the second (post-bootstrap) build, for better test coverage and since that's required after the merge from master.
* Make `init-tools.sh` actually fail in more failure cases.

Fixes #628 and #590.

Thanks, @akoeplinger!

### CODE REVIEW

File: init-tools.sh [hunk 1]
```diff
@@ -1,5 +1,7 @@
 #!/usr/bin/env bash
 
+set -e
```
Comment (cdmihai) [inline init-tools.sh]: Propagate back to corefx? :)

**Comment by cdmihai** [comment]
LGTM, hopefully the CI passes.

File: init-tools.sh (hunk 1 above)
Comment (rainersigwald) [inline init-tools.sh]: Smart. https://github.com/dotnet/corefx/pull/8621

**Comment by rainersigwald** [comment]
Ok, I don't understand these test failures. Neither happens on my machines

OSX: `dotnet restore` fails with a bunch of errors like:
```
EXEC : error : System.Reflection 4.0.10 provides a compile-time reference assembly for System.Reflection on .NETFramework,Version=v4.6, but there is no run-time assembly compatible with win7-x86. [/Users/dotnet-bot/j/workspace/Microsoft_msbuild/_OSX_prtest/targets/DeployDependencies.proj]
```

Ubuntu: more
```
/mnt/j/workspace/Microsoft_msbuild/_Ubuntu_prtest/Samples/PortableTask/PortableTask.csproj : error MSB4057: The target "Build" does not exist in the project.
```

@cdmihai Any ideas? Can *you* build these changes?

**Comment by cdmihai** [comment]
@dotnet-bot test this please

**Comment by cdmihai** [comment]
It builds on my OSX machine. Quite weird ...

**Comment by cdmihai** [comment]
@dotnet-bot test this please

**Comment by cdmihai** [comment]
@dotnet-bot test Ubuntu Build please

**Comment by rainersigwald** [comment]
@dotnet-bot test this please

**Comment by rainersigwald** [comment]
@dotnet-bot test this please

**Comment by rainersigwald** [comment]
@dotnet-bot test OSX Build please

**Comment by rainersigwald** [comment]
@dotnet-bot test OSX Build please

**Comment by rainersigwald** [comment]
Woo hoo! Looks like this is gated behind #659, then--when that's in I'll remove the merge and I'll feel good about pushing this.

**Comment by cdmihai** [comment]
:shipit:

**Pull request merged**